### PR TITLE
Simplify ERC1404 interface + many other cleanups

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/Auth.sol [hunk 1]
```diff
@@ -9,6 +9,12 @@ import {IAuth} from "src/interfaces/IAuth.sol";
 contract Auth is IAuth {
     mapping(address => uint256) public wards;
 
+    /// @dev Check if the msg.sender has permissions
```
Comment (hieronx) [inline src/Auth.sol:12]: More consistent to have modifier before functions

File: src/CentrifugeRouter.sol [hunk 2]
```diff
@@ -116,7 +116,7 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
     function unlockDepositRequest(address vault, address receiver) external payable protected {
         address initiator = _initiator();
         uint256 lockedRequest = lockedRequests[initiator][vault];
-        require(lockedRequest > 0, "CentrifugeRouter/user-has-no-locked-balance");
+        require(lockedRequest != 0, "CentrifugeRouter/no-locked-balance");
```
Comment (hieronx) [inline src/CentrifugeRouter.sol:119]: More gas efficient to use `!= 0` rather than `> 0`

File: src/token/Tranche.sol [hunk 3]
```diff
@@ -27,17 +27,13 @@ import {BitmapLib} from "src/libraries/BitmapLib.sol";
...
-    uint8 internal constant MAX_DECIMALS = 18;
-
     /// @inheritdoc ITranche
     address public hook;
 
     /// @inheritdoc IERC7575Share
     mapping(address asset => address) public vault;
 
-    constructor(uint8 decimals_) ERC20(decimals_) {
-        require(decimals_ <= MAX_DECIMALS, "Tranche/too-many-decimals");
```
Comment (hieronx) [inline src/token/Tranche.sol:39]: Already checked in pool manager

File: src/token/Tranche.sol [hunk 4]
```diff
@@ -148,15 +144,10 @@ contract Tranche is ERC20, ITranche {
     }
 
     /// @inheritdoc IERC1404
-    function messageForTransferRestriction(uint8 restrictionCode) public pure returns (string memory) {
+    function messageForTransferRestriction(uint8 restrictionCode) external pure returns (string memory) {
         return restrictionCode == SUCCESS_CODE_ID ? SUCCESS_MESSAGE : ERROR_MESSAGE;
     }
 
-    /// @inheritdoc IERC1404
-    function SUCCESS_CODE() public pure returns (uint8) {
```
Comment (hieronx) [inline src/token/Tranche.sol:156]: Not actually needed for ERC1404 I realized

File: src/libraries/CastLib.sol [hunk 5]
```diff
@@ -12,12 +12,6 @@ library CastLib {
         return bytes32(bytes(source));
     }
 
-    /// @dev Adds zero padding
-    function toBytes128(string memory source) internal pure returns (bytes memory) {
```
Comment (hieronx) [inline src/libraries/CastLib.sol:16]: Only used by tests so moved inline there

**Review by ilinzweilin — APPROVED** [review]

**Pull request merged**